components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/util/Oauth2ScopeUtils.java

```diff
@@ -338,7 +337,7 @@ public static String[] getRequestedScopes(String[] scopes) {
             return ArrayUtils.EMPTY_STRING_ARRAY;
         }
         for (String scope : scopes) {
-            if (scope.startsWith(INTERNAL_SCOPE_PREFIX) || scope.equalsIgnoreCase(SYSTEM_SCOPE)) {
```

Comment:
Are there any usages of this method outside of oauth component?
If not those methods might rely on this to filter the SYSTEM special scope

Reply:
No. It's used only in one place and it's modified in this PR. Checked by searching over the repo.